### PR TITLE
APP-1928: create release entity + metadata

### PR DESCRIPTION
**Pull request by Rekard0**
## Description

Task: [APP-1928](https://aragonassociation.atlassian.net/browse/APP-1928)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-1928]: https://aragonassociation.atlassian.net/browse/APP-1928?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/subgraph/src/plugin/pluginRepo.ts [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
   entity.pluginRepo = event.address.toHexString();
-  entity.release = event.params.release;
+  entity.release = pluginReleaseId;
   entity.build = event.params.build;
   entity.pluginSetup = pluginSetupId;
-  entity.metadata = event.params.buildMetadata;
+  entity.metadata = event.params.buildMetadata.toString();
   entity.save();
 
   let pluginSetupEntity = PluginSetup.load(pluginSetupId);
```
Comment (novaknole) [inline packages/subgraph/src/plugin/pluginRepo.ts:32]: Let's move this on top(before creating the `PluginVersion`) just for better predictability.

Comment (Rekard0) [reply packages/subgraph/src/plugin/pluginRepo.ts:32]: amended. thanks

File: packages/subgraph/src/plugin/pluginRepo.ts [hunk 2]
```diff
@@ -18,3 +35,25 @@ export function handleVersionCreated(event: VersionCreated): void {
...
+  let pluginRelease = event.params.release;
+  let releaseMetadata = event.params.releaseMetadata.toString();
+
+  let pluginReleaseEntityId = pluginRepoId
+    .concat('_')
+    .concat(pluginRelease.toString());
+
+  let pluginReleaseEntity = PluginRelease.load(pluginReleaseEntityId);
+  if (!pluginReleaseEntity) {
+    pluginReleaseEntity = new PluginRelease(pluginReleaseEntityId);
```
[the remark this answers is not in the document]
Comment (Rekard0) [reply packages/subgraph/src/plugin/pluginRepo.ts:52]: on the contract we have:
```
event VersionCreated(
        uint8 release,
        uint16 build,
        address indexed pluginSetup,
        bytes buildMetadata
    );
ReleaseMetadataUpdated(uint8 release, bytes releaseMetadata)
```
if it is the first time:
so first `VersionCreated` will be emitted, from this i have `release` and `build`
then `ReleaseMetadataUpdated` will be emitted, from this i have the `release`
on subgraph; we create `PluginVersion`, we create pluginReleaseId and we assign the id to the field of `release` on PluginVersion, even though there is no `PluginRelease` stored on the db yet, bcoz what it need at this point in time is an string id.
And once the `PluginRelease` got created in the future the relationship also will be established

[the remark this answers is not in the document]
Comment (novaknole) [reply packages/subgraph/src/plugin/pluginRepo.ts:53]: Thank you!

**Review by novaknole — APPROVED** [review]

File: packages/subgraph/src/plugin/pluginRepo.ts [hunk 3]
```diff
@@ -1,15 +1,32 @@
...
+  let pluginReleaseId = pluginRepoId.concat('_').concat(pluginRelease);
+
+  let entity = new PluginVersion(pluginVersionId);
   entity.pluginRepo = event.address.toHexString();
-  entity.release = event.params.release;
+  entity.release = pluginReleaseId;
   entity.build = event.params.build;
   entity.pluginSetup = pluginSetupId;
-  entity.metadata = event.params.buildMetadata;
+  entity.metadata = event.params.buildMetadata.toString();
```
[the remark this answers is not in the document]
Comment (Rekard0) [reply packages/subgraph/src/plugin/pluginRepo.ts:36]: we keep it as string for now, and we will tackle this in another task

File: packages/subgraph/src/plugin/pluginRepo.ts [hunk 4]
```diff
@@ -18,3 +35,25 @@ export function handleVersionCreated(event: VersionCreated): void {
     pluginSetupEntity.save();
   }
 }
+
+export function handleReleaseMetadataUpdated(
+  event: ReleaseMetadataUpdated
+): void {
+  let pluginRepoId = event.address.toHexString();
+  let pluginRelease = event.params.release;
+  let releaseMetadata = event.params.releaseMetadata.toString();
```
[the remark this answers is not in the document]
Comment (Rekard0) [reply packages/subgraph/src/plugin/pluginRepo.ts:45]: we keep it as string for now, and we will tackle this in another task

File: packages/subgraph/src/plugin/pluginRepo.ts [hunk 5]
```diff
@@ -1,20 +1,60 @@
...
+
+  let pluginReleaseEntityId = pluginRepoId
+    .concat('_')
+    .concat(pluginRelease.toString());
+
+  let pluginReleaseEntity = PluginRelease.load(pluginReleaseEntityId);
```
Comment (mathewmeconry) [inline packages/subgraph/src/plugin/pluginRepo.ts:51]: no need to load first and then calling `new`. As I found out subgraph merges the entities if you create a new one and the same exists already 
https://thegraph.com/docs/en/developing/assemblyscript-api/#updating-existing-entities

**Review by mathewmeconry — APPROVED** [review]

File: packages/subgraph/src/plugin/pluginRepo.ts (hunk 5 above)
Comment (Rekard0) [reply packages/subgraph/src/plugin/pluginRepo.ts:51]: nice, since we use load everywhere i guess we can go with load.

**Review by novaknole — APPROVED** [review]

**Pull request merged**